codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/CloudFrontKVSSigV4a.java

```diff
@@ -0,0 +1,145 @@
+import software.amazon.smithy.go.codegen.integration.ConfigField;
+import software.amazon.smithy.go.codegen.integration.ConfigFieldResolver;
+import software.amazon.smithy.go.codegen.integration.GoIntegration;
+import software.amazon.smithy.go.codegen.integration.MiddlewareRegistrar;
+import software.amazon.smithy.go.codegen.integration.RuntimeClientPlugin;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.model.traits.AuthTrait;
+import software.amazon.smithy.utils.ListUtils;
+import software.amazon.smithy.utils.SetUtils;
+
+/**
+ * This integration configures the CloudFront Key Value Store client for Signature Version 4a
+ */
+public class CloudFrontKVSSigV4a implements GoIntegration {
+    /**
+     * Return true if service is CFKVS.
+     *
+     * @param model   is the generation model.
+     * @param service is the service shape being audited.
+     */
+    private static boolean isCFKVSService(Model model, ServiceShape service) {
+        final String sdkId = service.expectTrait(ServiceTrait.class).getSdkId();
+        final String serviceId = sdkId.replace("-", "").replace(" ", "").toLowerCase();
+        return serviceId.equalsIgnoreCase("cloudfrontkeyvaluestore");
+    }
+
+    private final List<RuntimeClientPlugin> runtimeClientPlugins = new ArrayList<>();
+
+
+    @Override
+    public List<RuntimeClientPlugin> getClientPlugins() {
+        return runtimeClientPlugins;
+    }
+
+    @Override
+    public Model preprocessModel(Model model, GoSettings settings) {
+        ServiceShape service = settings.getService(model);
+        if (!isCFKVSService(model, service)) {
+            return model;
+        }
+
+        if (settings.getService(model).hasTrait(SigV4ATrait.class)) {
+            return model;
+        }
+
+        var v4a = SigV4ATrait.builder()
+                .name(service.expectTrait(SigV4Trait.class).getName())
+                .build();
+
+        return model.toBuilder()
+                .addShape(
+                        service.toBuilder()
+                                .addTrait(v4a)
+                                .addTrait(new AuthTrait(SetUtils.of(SigV4ATrait.ID, SigV4Trait.ID)))
+                                .build()
+                )
+                .build();
+    }
+
+    @Override
+    public void processFinalizedModel(GoSettings settings, Model model) {
+        if (!isCFKVSService(model, model.expectShape(settings.getService(), ServiceShape.class))) {
+            return;
+        }
+        runtimeClientPlugins.add(
+                RuntimeClientPlugin.builder()
+                    .configFields(
+                        ListUtils.of(
+                                ConfigField.builder()
+                                        .name(AwsSignatureVersion4aUtils.V4A_SIGNER_INTERFACE_NAME)
+                                        .type(SymbolUtils.createValueSymbolBuilder(
```

Comment:
nonblocking, remember that `SymbolUtils.buildPackageSymbol(String)` is now available (prefer usage via static import)